plugins/inputs/cloudwatch_metric_streams/README.md

```diff
@@ -0,0 +1,126 @@
+# CloudWatch Metric Streams Input Plugin
+
+The CloudWatch Metric Streams plugin is a service input plugin that
+listens for metrics sent via HTTP and performs the required
+processing for
+[Metric Streams from AWS](#troubleshooting-documentation).
+
+For cost, see the Metric Streams example in
+[CloudWatch pricing](#troubleshooting-documentation).
+
+## Configuration
+
+```toml @sample.conf
+[[inputs.metric_streams]]
+  ## Address and port to host HTTP listener on
+  service_address = ":443"
+
+  ## Paths to listen to.
+  # paths = ["/telegraf"]
+
+  ## maximum duration before timing out read of the request
+  # read_timeout = "10s"
+
+  ## maximum duration before timing out write of the response
+  # write_timeout = "10s"
+
+  ## Maximum allowed http request body size in bytes.
+  ## 0 means to use the default of 524,288,000 bytes (500 mebibytes)
+  # max_body_size = "500MB"
+
+  ## Optional username and password to accept for HTTP basic authentication.
+  ## You probably want to make sure you have TLS configured above for this.
+  # basic_username = "foobar"
+  # basic_password = "barfoo"
+```
+
+## Metrics
+
+Metrics sent by AWS are Base64 encoded blocks of JSON data.
+There can be multiple blocks of JSON for each `data` field
+in a `record` and there can be multiple `records`.
+
+The metric when decoded may look like this:
+
+```json
+{
+    "metric_stream_name": "sandbox-dev-cloudwatch-metric-stream",
+    "account_id": "541737779709",
+    "region": "us-west-2",
+    "namespace": "AWS/EC2",
+    "metric_name": "CPUUtilization",
+    "dimensions": {
+        "InstanceId": "i-0efc7ghy09c123428"
+    },
+    "timestamp": 1651679580000,
+    "value": {
+        "max": 10.011666666666667,
+        "min": 10.011666666666667,
+        "sum": 10.011666666666667,
+        "count": 1
+    },
+    "unit": "Percent"
+}
+```
```

Comment:
I personally think so because the actual data model was not readily available by AWS, I had to send data and the decode it.

So this JSON is just the metric itself, skipping the other details that outlined by AWS in their documentation.

Reply:
Okay, but what about the references to  `data`, `record` or `records` then?

Reply:
They are outlined here: https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html

Do you think I should add this here?

There is also a full sample message in the tests, but it's a bit too long to add to the README.

Reply:
I would mention this sample JSON is what comes from the decoded `data` field. Or didn't I understand the docs correctly?

Reply:
No that's right, so a request from AWS looks like:

```json
{
  "requestId": "ed4acda5-034f-9f42-bba1-f29aea6d7d8f",
  "timestamp": 1578090901599,
  "records": [
   {
    "data": {
```

And the `data` is one Base64 encoded string, that is actually multiple JSON blocks.

I think it might be a bit too much developer information to add all of that, but I will update the README to include that the JSON is first Base64 encoded within `data`.